### ISSUE TITLE
keras.ops.pad works incorrectly with pytorch backend

### ISSUE THREAD

**Issue by caiuspetronius** [keras-team/keras#20347]
When used with 5D tensors and 'symmetric' or 'reflect' padding mode and pad_width = ( ( 0, 0), (1, 1), (1, 1), (1, 1), (0, 0 )), to pad only 3 spatial dimensions, the ops.pad works as expected with TF and JAX backends, but returns the error when used with torch: "Only 2D, 3D, 4D, 5D padding with non-constant padding are supported for now". If the pad_width is truncated to ( ( 0, 0), (1, 1), (1, 1), (1, 1) ) the error doesn't appear, but in this case 2 last spatial dimensions + channels dimension are padded instead of the 3 spatial dimensions. If the pad_width is changed to ( (1, 1), (1, 1), (1, 1), (0, 0) ) the same error is returned. TF 2.17.0, Keras 3.6.0, torch 2.4.1.

**Comment by Grvzard** [comment]
refer to:
https://github.com/keras-team/keras/blob/8e7843c9f8a3b3298730ab993fd6e6d7bf5c2b1c/keras/src/backend/torch/numpy.py#L1102-L1105

> If the pad_width is truncated to ( ( 0, 0), (1, 1), (1, 1), (1, 1) ) the error doesn't appear ...

It's truly a bug in keras. I will submit a PR later. Thank you for pointing it out!

**Comment by caiuspetronius** [comment]
You are welcome, thanks!

**Comment by fchollet** [comment]
PR has been merged -- thanks!

**Comment by caiuspetronius** [comment]
Hi fchollet, how can I check if this was fixed? I tried installing keras-nightly, but the bug is still there.

**Comment by Grvzard** [comment]
> but the bug was still there.

It's a limitation in pytorch, not keras. You can check the official documentation linked below.
https://github.com/keras-team/keras/blob/8e7843c9f8a3b3298730ab993fd6e6d7bf5c2b1c/keras/src/backend/torch/numpy.py#L1102-L1105

**Comment by caiuspetronius** [comment]
I'm not sure what you mean by "not a limitation of keras". It is certainly a limitation, because any keras network with non-constant padding won't be compatible with pytorch until this issue is resolved. The constant padding works, btw, even though torch padding still works only n the last 3 dimensions. Obviously, the appropriate tensor permutation has to be done before the padding is implemented in torch. If it was implemented correctly for 'constant' padding then why the same can't be done for the other modes?
